File: backend/Services/Parsers/MoviemojiParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class MoviemojiParser : BasicScoreResultParser
    {
        private ILogger<MoviemojiParser> _logger;

        public MoviemojiParser(ILogger<MoviemojiParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Moviemoji";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} #\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => "Play here: ";
        public override string Url => "https://moviedle.xyz/moviemoji/";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace(ExtraContent, string.Empty).Replace(Url, string.Empty).Trim();
        }
    }
}

File: lambdas/src/parsers/moviemoji-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class MoviemojiParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Moviemoji';
  readonly helpText = null;
  protected readonly extraContent = 'Play here: ';
  readonly url = 'https://moviedle.xyz/moviemoji/';
  
  // Matches "Moviemoji #489 X/8" or "Moviemoji #465 1/8"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace(this.extraContent ?? '', '')
      .replace(this.url ?? '', '')
      .trim();
  }
}
